scio-tensorflow/src/main/scala/com/spotify/scio/values/TFSCollectionFunctions.scala

```diff
@@ -126,6 +132,61 @@ class TensorFlowSCollectionFunctions[T: ClassTag](@transient val self: SCollecti
   }
 }
 
+class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
+
+  /**
+   * Save this SCollection of [[org.tensorflow.example.Example]] as a TensorFlow TFRecord file.
+   *
+   * @param featureSpec feature spec for the Examples. This collection must contain
+   *                    a single element.
+   * @param featureSpecPath path to save the feature specification to, by default it will be
+   *                        `${path}/.feature_spec`
+   *
+   * @group output
+   */
+  def saveAsTfExampleFile(path: String,
+                          featureSpec: SCollection[Seq[String]],
+                          suffix: String = ".tfrecords",
+                          compressionType: CompressionType = CompressionType.NONE,
+                          numShards: Int = 0,
+                          @Nullable featureSpecPath: String = null)
+                         (implicit ev: T <:< Example)
+  : (Future[Tap[Example]], Future[Tap[String]]) = {
+    require(featureSpec != null, "Feature spec can't be null")
+    require(path != null, "Path can't be null")
+    val _featureSpecPath =
+      Option(featureSpecPath).getOrElse(path.replaceAll("\\/+$", "") + "/.feature_spec")
```

Comment:
Does `.dotfile` play well with GCS or S3?

Reply:
What do you mean? it would be another object; unless you use some kinda of abstraction on top of it, like HadoopFS which would actually assume `.`|`_` is hidden.

Reply:
I mean if filenames like `.feature_spec` would cause problem on some file systems. Probably not?

Reply:
Is fine on GCS, HDFS. S3 should be fine too http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html